node_modules/gh-groups/tests/util.js

```diff
@@ -20,6 +20,57 @@ var _ = require('lodash');
 var assert = require('assert');
 
 /**
+ * Assert that the groups of a user can be retrieved
+ *
+ * @param  {RestClient}         client                  The REST client to make the request with
+ * @param  {Number}             id                      The id of the user to get the memberships of
+ * @param  {Number}             [limit]                 The amount of groups that should be returned
+ * @param  {Number}             [offset]                The paging number of the results to retrieve
+ * @param  {User[]}             [expectedMemberships]   The expected memberships
+ * @param  {Function}           callback                Standard callback function
+ * @param  {OrgUnit}            callback.memberships    The members of the group
+ * @throws {AssertionError}                             Error thrown when an assertion failed
+ */
+var assertGetUserMembershipsEquals = module.exports.assertGetUserMembershipsEquals = function(client, id, limit, offset, expectedMemberships, callback) {
+    client.group.getUserMemberships(id, limit, offset, function(err, memberships) {
+        assert.ok(!err);
+        assert.ok(memberships);
+        assert.ok(memberships.results);
+
+        if (expectedMemberships) {
+            assert.strictEqual(memberships.results.length, expectedMemberships.length);
+            _.each(memberships.results, function(membership) {
+                assert.ok(_.contains(expectedMemberships, membership.id));
+            });
+        }
```

Comment:
We don't have to put a conditional on `expectedMemberships` since this function always tests for equality, and we should do an `assert.deepEqual` on a sorted copy of the ids of the expected and actual memberships list.

The current algorithm I think currently passes with `[1, 1] === [1, 2]`